azkeys/coseSigner.go

```diff
@@ -89,6 +90,7 @@ func (f *KeyVaultCoseSignerFactory) NewIdentifiableCoseSigner(ctx context.Contex
 // KeyVaultCoseSigner is the azure keyvault client for interacting with keyvault keys
 // using a cose.Signer interface
 type KeyVaultCoseSigner struct {
+	ctx                context.Context
```

Comment:
Why do you want to store this here? Is it because the interface methods don't accept a context, and therefore we can't cancel them? I've pasted some guidance from the docs below ... but it doesn't feel quite as applicable as instances of this factory should be transient. If we keep this, we could add commentary to this effect on the struct. What do you think?

---
> The [documentation for context](https://go.dev/pkg/context/) states:
    Contexts should not be stored inside a struct type, but instead passed to each function that needs it.
    
https://go.dev/blog/context-and-structs